test/test_integration.py

```diff
@@ -49,6 +50,13 @@ def test_spin_backend() -> None:
     connectivity = {0: [1, 2], 1: [0], 2: [0, 3], 3: [2]}
 
     qc.route(router=RoutingChecker(connectivity))
+
+    # Check whether the gates in the circuit match the native gate set of the backend
+
+    native_gate_set = ["Rx", "Ry", "H", "Rz", "CR", "CRk", "SWAP", "CNOT", "measure"]
+
+    qc.check(checker=NativeGateChecker(native_gate_set))
```

Comment:
This compilation pass should be placed at the end, _i.e._ after the McKay decomposition pass.

```suggestion
    # Validate that the fully decomposed circuit consists of gates that are in the native gate set.
    qc.validate(validator=NativeGateValidator(native_gate_set))
```